apis/networking/v1beta1/apiproduct_types.go

```diff
@@ -60,35 +60,22 @@ type ProductInformation struct {
 	Owner       string `json:"owner"`
 }
 
-type Environment struct {
-	Name              string              `json:"name"`
-	Hosts             []string            `json:"hosts"`
-	TLSConfig         *TLSConfig          `json:"tlsConfig,omitempty"`
-	CredentialSources []*CredentialSource `json:"credentialSources"`
-	BackendServers    []*BackendServer    `json:"backendServers"`
-}
-
-type CredentialSource struct {
-	Name              string                        `json:"name"`
-	APIKeyAuth        *APIKeyAuthCredentials        `json:"apiKeyAuth,omitempty"`
-	OpenIDConnectAuth *OpenIDConnectAuthCredentials `json:"openIDConnectAuth,omitempty"`
-}
-
 type APIKeyAuthCredentials struct {
 	LabelSelectors map[string]string `json:"labelSelectors"`
 }
 
-type OpenIDConnectAuthCredentials struct {
-	Endpoint string `json:"endpoint"`
+type Routing struct {
+	Hosts  []string `json:"hosts"`
+	Expose bool     `json:"expose"`
 }
 
-type BackendServer struct {
-	API         string      `json:"API"`
-	Destination Destination `json:"destination"`
+type OpenIDConnectAuthCredentials struct {
+	Endpoint string `json:"endpoint"`
 }
 
 type Destination struct {
```

Comment:
`Destination` object type belong to the API object type.

Good to have the destination in the API, instead of the APIProduct